pkg/istio/rbac/common/servicerolebinding.go

```diff
@@ -43,17 +51,28 @@ func GetServiceRoleBindingSpec(k8sRoleName string, members []*zms.RoleMember) (*
 
 		//TODO: handle member.Expiration for expired members, for now ignore expiration
 
-		memberName, err := parseMemberName(member)
+		spiffeName, requestAuthPrincipal, err := parseMemberName(member)
 		if err != nil {
 			log.Warningln(err.Error())
 			continue
 		}
 
-		subject := &v1alpha1.Subject{
-			User: memberName,
+		// Spiffe and request auth principal subjects MUST be separate or else
+		// the user needs to provide both the certificate and the jwt token. If
+		// one subject is used, the source.principal and request.auth.principal
+		// in the envoy rbac is grouped together into one id principal array as
+		// opposed to being separated to allow either to go through.
+		spiffeSubject := &v1alpha1.Subject{
+			User: spiffeName,
+		}
+
+		requestAuthPrincipalSubject := &v1alpha1.Subject{
```

Comment:
could we have the jwt subject added based on a cmd line flag (e.g. `--enable-origin-jwt-subject` ) or a better name, so that we can turn on/off quickly. In this case, `parseMemberName` could be separated and renamed to something like `memberToSpiffe` and the jwt one to`memberToOriginSubject`. What do you think?

Reply:
I agree we should add the feature flag in case we want to roll back. It's a little bit tricky because this function does not have access to much besides its parameters. I am thinking of two ways of doing this:
1. Add the flag to the provider object which will pass it as a parameter to GetServiceRoleBindingSpec function
2.  Introduce the feature flag global object used in kubernetes. This might be overkill as we only have one flag for now but we could potentially have more in the future.

Reply:
+1 to feature flag it

Reply:
creating a global object in k8s seems overkill for feature flag.

Reply:
@prabushyam @prabushyam I have added the flag and also separated the functions. Please review.